### PR TITLE
fix: text styles when using CJK IME

### PR DESCRIPTION
**Pull request by zqran**
close: #1553

### CODE REVIEW

**Comment by Flrande** [comment]
Currently, all text is inserted without attributes by default, and if we wish to maintain formatting, we must handle it externally, such as using the [virgoInput handler](https://github.com/toeverything/blocksuite/blob/29f3a9a1cf18f592cb4427b1a78d5e2432abd8a5/packages/blocks/src/__internal__/rich-text/rich-text.ts#L72). (The changes in this pr now will break this design)

To address this issue in a more reasonable manner, I propose adding handlers for IME, such as `virgoCompositionStart` and `virgoCompositionEnd`. By doing so, we can keep the system simple and customizable. :)

**Review by Flrande — CHANGES_REQUESTED** [review]
https://github.com/toeverything/blocksuite/pull/1636#issuecomment-1463205297

File: packages/blocks/src/__internal__/rich-text/rich-text.ts [hunk 1]
```diff
@@ -98,6 +98,31 @@ export class RichText extends NonShadowLitElement {
...
+        if (!vRange || vRange.length !== 0) {
+          return false;
+        }
+
+        const index = vRange.index;
+        if (index >= 0 && data) {
+          const delta = vEditor.getDeltaByRangeIndex(index);
+          const attributes = delta?.attributes ?? {};
+          vEditor.insertText(vRange, data, attributes);
+          vEditor.slots.updateVRange.emit([
```
Comment (Flrande) [inline packages/blocks/src/__internal__/rich-text/rich-text.ts:115]: You can use `setVRange` here,

**Comment by zqran** [comment]
> Currently, all text is inserted without attributes by default, and if we wish to maintain formatting, we must handle it externally, such as using the [virgoInput handler](https://github.com/toeverything/blocksuite/blob/29f3a9a1cf18f592cb4427b1a78d5e2432abd8a5/packages/blocks/src/__internal__/rich-text/rich-text.ts#L72). (The changes in this pr now will break this design)
> 
> To address this issue in a more reasonable manner, I propose adding handlers for IME, such as `virgoCompositionStart` and `virgoCompositionEnd`. By doing so, we can keep the system simple and customizable. :)

Thanks for the detailed explanation, such a design is awesome.

**Review by Flrande — CHANGES_REQUESTED** [review]
I recommend using `setVRange`

File: packages/blocks/src/__internal__/rich-text/rich-text.ts (hunk 1 above)
Comment (zqran) [reply packages/blocks/src/__internal__/rich-text/rich-text.ts]: 👌

**Review by Flrande — APPROVED** [review]

**Comment by Flrande** [comment]
Thank for your work!

**Comment by lawvs** [comment]
This incorrect fix will result in inconsistent input states between using and not using IME.

https://user-images.githubusercontent.com/18554747/224262789-ceedf814-9a16-4cd9-9559-5a25fa1d974f.mov

**Comment by Flrande** [comment]
It seem that it need to compute conditional branch like `virgoInput`. 🤔

**Comment by zqran** [comment]
I'll try to fix it.

**Comment by Flrande** [comment]
> I'll try to fix it.

https://github.com/toeverything/blocksuite/blob/a86b71fbe4dab0a34d87e62df297163f5949a651/packages/blocks/src/__internal__/rich-text/rich-text.ts#L82-L89
Maybe it can help you.

**Pull request merged**